fuel-block-producer/src/block_producer.rs

```diff
@@ -194,17 +194,11 @@ impl Producer {
     }
 
     fn previous_block_info(&self, height: BlockHeight) -> Result<PreviousBlockInfo> {
+        // TODO: It is not guaranteed that the genesis height is `0` height. Update the code to
```

Comment:
ahh good catch, since we allow setting any arbitrary starting block height in the chainspec.